internal/maintenance/reindexer.go

```diff
@@ -105,31 +105,48 @@ func (s *Reindexer) Start(ctx context.Context) error {
 		s.Logger.DebugContext(ctx, s.Name+logPrefixRunLoopStarted)
 		defer s.Logger.DebugContext(ctx, s.Name+logPrefixRunLoopStopped)
 
-		// On each run, we calculate the new schedule based on the previous run's
-		// start time. This ensures that we don't accidentally skip a run as time
-		// elapses during the run.
-		lastRunAt := time.Now().UTC()
+		nextRunAt := s.Config.ScheduleFunc(time.Now().UTC())
 
-		for {
-			nextRunAt := s.Config.ScheduleFunc(lastRunAt)
+		s.Logger.InfoContext(ctx, s.Name+": Scheduling first run", slog.Time("next_run_at", nextRunAt))
```

Comment:
Added this log line so it's easier to observe the right thing happening from the test suite.